evennia/contrib/wilderness.py

```diff
@@ -0,0 +1,490 @@
+        """Returns True if location is valid and can be walked to.
+
+        Args:
+            wilderness: the wilderness script
+            location ((int, int)): the location to check as (x, y) tuple.
+        """
+        x, y = location
+        if x < 0:
+            return False
+        if y < 0:
+            return False
+
+        return True
+
+    def get_location_name(self, location):
+        """Returns a name for the position at location.
+
+        Args:
+            location ((int, int)): the location as (x, y) tuple.
+        """
+        return "The wilderness"
+
+
+def create_wilderness(name="default", mapprovider=None):
+    """Creates a new wilderness map.
+
+    Args:
+        name (str, optional): the name to use for that wilderness map
+        mapprovider (WildernessMap instance, optional): an instance of a
+            WildernessMap class (or subclass) that will be used to provide the
+            layout of this wilderness map. If none is provided, the default
+            infinite grid map will be used.
+
+    """
+    if WildernessScript.objects.filter(db_key=name).exists():
+        # Don't create two wildernesses with the same name
+        return
+
+    if not mapprovider:
+        mapprovider = WildernessMapProvider()
+    script = create_script(WildernessScript, key=name)
+    script.db.provider = mapprovider
+
+
+def enter_wilderness(obj, location=(0, 0), name="default"):
+    """Moves obj into the wilderness. The wilderness needs to exist first.
+    Returns True on success.
+
+    Args:
+        obj (object): the object to move into the wilderness
+        location ((int, int), optional): the location to move obj to into the
+            wilderness. If not provided, defaults (0, 0)
+        name (str, optional): name of the wilderness map, if not using the
+            default one
+    """
+    script = WildernessScript.objects.get(db_key=name)
+    if script.is_valid_location(location):
+        script.move_obj(obj, location)
+        return True
+    else:
+        return False
+
+
+def get_new_location(location, direction):
+    """Returns the coordinates of direction applied to location.
+
+    Args:
+        location: tuple of (x, y)
+        direction: a direction string (like "northeast")
+    """
+    x, y = location
+
+    if direction in ("north", "northwest", "northeast"):
+        y += 1
+    if direction in ("south", "southwest", "southeast"):
+        y -= 1
+    if direction in ("northwest", "west", "southwest"):
+        x -= 1
+    if direction in ("northeast", "east", "southeast"):
+        x += 1
+
+    return (x, y)
+
+
+class WildernessScript(DefaultScript):
+
+    def at_script_creation(self):
+        self.persistent = True
+
+        # Store the location of every item that is inside the wilderness
+        # Key: object, Value: (x, y)
+        self.db.itemlocations = {}
+
+        # Store the rooms that are used as views into the wilderness
```

Comment:
That's a good idea! Currently that's not possible as the rooms would be moved away to storage when they're not needed anymore. But adding a "keep" flag could fix that.